pkg/openai/main.go

```diff
@@ -104,22 +109,53 @@ func (c *Client) sendReq(reqURL, method, contentType string, data io.Reader) ([]
 		req.Header.Add("OpenAI-Organization", c.Org)
 	}
 	http.DefaultClient.Timeout = reqTimeout
+
 	res, err := c.HTTPClient.Do(req)
 	if res != nil && res.Body != nil {
 		defer res.Body.Close()
 	}
 	if err != nil || res == nil {
-		err = fmt.Errorf("error occurred: %v, while calling URL: %s", err, reqURL)
-		return nil, err
+		logger.Warn("Failed to call OpenAI", zap.Error(err))
+		return nil, errmsg.AddMessage(
+			fmt.Errorf("failed to call OpenAI: %w", err),
+			"Failed to call OpenAI's API.",
+		)
 	}
+
 	respBody, err := io.ReadAll(res.Body)
 	if err != nil {
 		return nil, err
 	}
-	if res.StatusCode != http.StatusOK {
-		err = fmt.Errorf("non-200 status code: %d, while calling URL: %s, response body: %s", res.StatusCode, reqURL, respBody)
-		return nil, err
+
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
```

Comment:
ℹ️ I used a broader range of statuses as there's no contract around OpenAI response codes.